crates/milli/src/search/facet/filter.rs

```diff
@@ -434,36 +466,56 @@ impl<'a> Filter<'a> {
-                            .union()
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
+                    Some(features) if features.is_filterable() => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        if let Some(fid) = field_ids_map.id(fid.value()) {
+                            els.iter()
+                                .map(|el| Condition::Equal(el.clone()))
+                                .map(|op| {
+                                    Self::evaluate_operator(
+                                        rtxn, index, fid, universe, &op, &features,
+                                    )
+                                })
+                                .union()
+                        } else {
+                            Ok(RoaringBitmap::new())
+                        }
+                    }
+                    _ => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        Err(fid.as_external_error(FilterError::AttributeNotFilterable {
+                            attribute: fid.value(),
+                            filterable_fields: filtered_matching_field_names(
+                                filterable_fields,
+                                &field_ids_map,
+                                &|features| features.is_filterable(),
+                            ),
+                        }))?
                     }
-                } else {
-                    Err(fid.as_external_error(FilterError::AttributeNotFilterable {
-                        attribute: fid.value(),
-                        filterable_fields: filterable_fields.clone(),
-                    }))?
                 }
             }
             FilterCondition::Condition { fid, op } => {
-                if crate::is_faceted(fid.value(), filterable_fields) {
-                    let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(fid.value()) {
-                        Self::evaluate_operator(rtxn, index, fid, universe, op)
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
+                    Some(features) if features.is_filterable() => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        if let Some(fid) = field_ids_map.id(fid.value()) {
+                            Self::evaluate_operator(rtxn, index, fid, universe, op, &features)
+                        } else {
+                            Ok(RoaringBitmap::new())
+                        }
+                    }
+                    _ => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        Err(fid.as_external_error(FilterError::AttributeNotFilterable {
```

Comment:
I thought this case was checked in `evaluate` already.

Do we need to keep the pre-check in `evaluate` then?

Reply:
> Do we need to keep the pre-check in evaluate then?
We could remove it, however, I feel the check is an early return error in case the last filter is wrong avoiding to evaluate the other filter for nothing, on the other side, we could replace each match case by `unreachable!` knowing that we already checked if the field is filterable but I am not sure if we want to add "possible" panics in the search 🤔 

I don't have a strong opinion here, I'll follow the call

Reply:
> I feel the check is an early return error in case the last filter is wrong avoiding to evaluate the other filter for nothing,

I don't feel that optimizing for the error case is right, especially when the evaluation of a filter is supposed to be a relatively quick operation (so that search is quick).

I'd rather we remove the pre-check then, if we are sure this won't create an observable behavior difference.